### PR TITLE
Refactor commercial-component

### PR DESCRIPTION
**Pull request by regiskuckaertz**
There was lots of useless work going on in this module and it was (to me at least) very hard to reason about.

It runs 10–20x faster than the previous version, but we're talking milliseconds here so it's not really a good selling point.

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js [hunk 1]
```diff
@@ -32,72 +24,54 @@ define([
...
+        };
+    }
+
+    function soulmatesGroupUrlBuilder(url) {
+        return function (params) {
+            return buildComponentUrl(url + params.soulmatesFeedName, params);
         };
+    }
+
+    function complexUrlBuilder(url, withT, withKeywords) {
```
Comment (uplne) [inline static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js]: What does it mean withT?

Comment (regiskuckaertz) [inline static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js]: `t` is the name of the parameter. I'll change it to `withSpecificID`, that is its real use.

File: static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js [hunk 2]
```diff
@@ -32,72 +24,54 @@ define([
...
+    map,
+    reduce
 ) {
+    var urlBuilders = {
+        bestbuy:        defaultUrlBuilder('money/bestbuys'),
+        soulmates:      defaultUrlBuilder('soulmates/mixed'),
+        soulmatesTest:  defaultUrlBuilder('soulmates-test/mixed'),
+        capiSingle:     defaultUrlBuilder('capi-single'),
+        capi:           defaultUrlBuilder('capi'),
+        paidforCard:    defaultUrlBuilder('paid'),
```
Comment (uplne) [inline static/src/javascripts/projects/common/modules/commercial/creatives/commercial-component.js]: Let's order this:

defaultUrlBuilder,
complexUrlBuilder,
bookUrlBuilder,
soulmatesGroupUrlBuilder

**Comment by johnduffell** [comment]
looks awesome to me, if it's good code we should get this out the door! :+1: if not close

**Pull request merged**